### ISSUE TITLE
Helm swoop without input, if nothing selected?

### ISSUE THREAD

**Issue by ReneFroger** [ShingoFukuyama/helm-swoop#39]
When my cursor is on a word, and I start Helm Swoop, I noticed that word where my cursor was placed, is putted in Helm swoop input. I would like to prevent that. 

But in a manner way if I selected nothing, there will be nothing putted into swoop.
But when I select a word or sentence, that will be putted in Helm swoop then. Is this possible?

And yes, I have read the manual before I asked that question. :smiley:

**Comment by tuhdo** [comment]
> But in a manner way if I selected nothing, there will be nothing putted into swoop.

You can customize `helm-swoop-pre-input-function` by deleting its value and change to: `(lambda () "")`. Although this is customizable, the way it displays is not friendly to easily customize. @ShingoFukuyama I suggest that you should change this variable to radio type that has the following options:

- symbol
- list
- sexp
- defun
- filename
- url
- email
- word
- sentence
- whitespace
- line
- number
- page
- custom function (allow user to enter a function name)[
- nil

Basically the options are things that `thing-at-point` accepts and you can remove options that you find irrelevant (i.e. page, line and number). `custom-function` is for user to assign a function name and `nil` means nothing.

> But when I select a word or sentence, that will be putted in Helm swoop then. Is this possible?

This is already implemented. If you have an active region, `helm-swoop` uses it.

**Comment by ShingoFukuyama** [comment]
Added a few examples on the document.
